lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -302,4 +288,47 @@ public function matching(Criteria $criteria)
 
         return new LazyCriteriaCollection($persister, $criteria);
     }
+
+    /**
+     * Resolves a magic method call to the proper existent method at `EntityRepository`.
+     *
+     * @param string $method    The method to call
+     * @param string $by        The property name used as condition
+     * @param array  $arguments The arguments to pass at method call
+     *
+     * @throws ORMException If the method called is invalid.
+     *
+     * @return mixed
+     */
+    private function resolveMagicCall($method, $by, array $arguments = [])
+    {
+        $argsCount = count($arguments);
+
+        if (0 === $argsCount) {
+            throw ORMException::findByRequiresParameter($method . $by);
+        }
+
+        $fieldName = lcfirst(\Doctrine\Common\Util\Inflector::classify($by));
+
+        if ($this->_class->hasField($fieldName) || $this->_class->hasAssociation($fieldName)) {
+            switch ($argsCount) {
+                case 1:
+                    return $this->$method(array($fieldName => $arguments[0]));
+
+                case 2:
+                    return $this->$method(array($fieldName => $arguments[0]), $arguments[1]);
+
+                case 3:
+                    return $this->$method(array($fieldName => $arguments[0]), $arguments[1], $arguments[2]);
+
+                case 4:
```

Comment:
Nope: already checked out locally and took ownership of the branch. Relax, sit back and have a 🍺